### ISSUE TITLE
Gluten will return wrong result when calling the df.describe() method

### ISSUE THREAD

**Issue by JkSelf** [oap-project/gluten#1354]
**Describe the bug**
Gluten will return wrong result or core dump with negative buffer size when calling df.describe() method.

**To Reproduce**
```
test("test df.describe() method") {
    val d = 3
    val df = Seq(d, d, d, d, d, d, d, d, d, d).toDF("IntegerCol")
    df.describe().show()
  }
```

**Comment by PHILO-HE** [comment]
It looks like a similar issue for testing "Gluten - describe" in GlutenDataFrameSuite. I just disabled it in https://github.com/oap-project/gluten/pull/984. Please help enable it after your fix. Thanks!

**Comment by JkSelf** [comment]
This is a bug fix for vanilla spark in spark 3.2 and 3.3. We have fixed it in [PR#40914](https://github.com/apache/spark/pull/40914) in spark 3.3. And this issue will be fixed after gluten upgrade the spark version to spark3.4.
